src/katgpucbf/xbgpu/correlation.py

```diff
@@ -73,13 +73,20 @@ class CorrelationTemplate:
         The number of time samples to be processed per frequency channel.
     """
 
-    def __init__(self, context: AbstractContext, n_ants: int, n_channels: int, n_spectra_per_heap: int) -> None:
+    def __init__(
+        self,
+        context: AbstractContext,
+        n_ants: int,
+        n_channels: int,
+        n_spectra_per_heap: int,
+        sample_bitwidth: int,
+    ) -> None:
         self.n_ants = n_ants
         self.n_channels = n_channels
         self.n_spectra_per_heap = n_spectra_per_heap
         self.n_baselines = self.n_ants * (self.n_ants + 1) // 2
 
-        self._sample_bitwidth = 8  # hardcoded to 8 for now, but 4 and 16 bits are also supported
+        self._sample_bitwidth = sample_bitwidth  # hardcoded to 8 upstream, but 4 and 16 bits are also supported
```

Comment:
It's probably me that wrote the comment, but I think claiming they're "supported" without having any corresponding tests is rather optimistic (I was probably thinking that TCC supports it, rather than this module).

For now I'd suggest raising ValueError if `sample_bitwidth != 8`. There is currently a check at engine.py:1024 that I think can be deleted from there and moved here.

Reply:
I'd also suggest turning it into a public member (no leading `_`) since you're now accessing it from outside the class.